### ISSUE TITLE
[a11y] RangeSlider keyboard focus does not follow mouse interaction

### ISSUE THREAD

**Issue by yjbanov** [flutter/flutter#173575]
## Context

* This issue is similar to the past issue we used to have with buttons: https://github.com/flutter/flutter/issues/158527
* This is a follow-up for this discussion: https://github.com/flutter/flutter/pull/172729#discussion_r2238000656
* This may be related or fixable together with https://github.com/flutter/flutter/issues/173574.

## Reproduction

1. In the Flutter repo, `cd dev/a11y_assessments`.
2. `flutter run`
3. Enable the additional use-cases using the switch in the top-right corner of the app.
4. Open the RangeSlider use-case.
5. Use the mouse or trackpad to move the left thumb.
6. Let go of the mouse, and use the ARROW RIGHT/LEFT keys to operate the range slider again.

## Expected behavior

The keyboard input focus should have moved to the thumb that was moved during step 5, and using the ARROW RIGHT/LEFT keys during step 6 should have moved that same thumb.

It should also be possible to switch focus between the left and the right thumb by clicking on the respective thumb. For example, if the left thumb currently has focus (use the TAB key to get it), the user should be able to click on the right thumb, and then operate it using the keyboard. This is currently not happening.

## Actual behavior

The thumb that was used by the mouse last does not react to the keyboard.
